### PR TITLE
Enable dependabot to automatically get dependency updates

### PR DESCRIPTION
**Pull request by MRamonLeon**
It would be great to have **Dependabot** configured to get dependencies updates filed automatically. This will help avoiding bugs or security vulnerabilities from old dependencies.

Some helpful documentation about Dependabot to fine adjust it if needed:

* Keeping your dependencies updated automatically: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/keeping-your-dependencies-updated-automatically
* Avoiding security vulnerabilities: https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/configuring-github-dependabot-security-updates
* Fine adjusting the configuration: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/configuration-options-for-dependency-updates

### CODE REVIEW

**Comment by MRamonLeon** [comment]
cc: @gboissinot @wolfs

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+  - package-ecosystem: "maven"
```
Comment (jglick) [inline .github/dependabot.yml:3]: Uh, not going to work when deps are in `build.gradle`.

AFAIK use of Dependabot on Gradle-based Jenkins plugins is untested and may or may not work sanely at all.

**Review by jglick — CHANGES_REQUESTED** [review]

**Review by wolfs — CHANGES_REQUESTED** [review]
AFAIK dependabot does not work with Gradle.

**Comment by jglick** [comment]
> dependabot does not work with Gradle

Acc. to https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/configuration-options-for-dependency-updates#package-ecosystem `gradle` is an option. Whether the updates it would offer would be at all useful is another question.

**Review by wolfs — APPROVED** [review]
If it works for Gradle, we can try it out. Is the dependabot configuration valid for a Gradle build, or is there something special which needs to be added? Can we test the integration on this PR already?

**Comment by jglick** [comment]
> Is the dependabot configuration valid for a Gradle build

If you changed where it says `maven` to `gradle`. :grin: 

> Can we test the integration on this PR already?

I am not aware of a way to test the integration—you just turn it on, and if you get a half dozen worthless PRs that you have to close, you tune it or turn it off.

**Comment by wolfs** [comment]
@MRamonLeon @jglick I merged the PR and am now getting PRs to update dependencies. I don't want to update the Spock dependency, since it requires a too high Groovy version: https://github.com/jenkinsci/gradle-plugin/pull/103
How can I disable dependabot updates for this dependency?

**Comment by jglick** [comment]
> How can I disable dependabot updates for this dependency?

https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/configuration-options-for-dependency-updates#ignore

**Pull request merged**